static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner.spec.js

```diff
@@ -0,0 +1,141 @@
+// @flow
+import fakeTemplate from 'lodash/utilities/template';
+import FakeMessage from 'common/modules/ui/message';
+import fakeMediator from 'lib/mediator';
+import { commercialFeatures } from 'commercial/modules/commercial-features';
+import { membershipEngagementBannerInit } from 'common/modules/commercial/membership-engagement-banner';
+
+jest.mock('lib/storage', () => ({
+    local: {
+        get: jest.fn(() => 10), // gu.alreadyVisited
+        set: jest.fn(),
+        isAvailable: jest.fn(),
+    },
+}));
+jest.mock('lib/url', () => ({
+    constructQuery: jest.fn(),
+}));
+jest.mock('lib/geolocation', () => ({
+    get: jest.fn(() => Promise.resolve('GB')),
+}));
+jest.mock('lib/mediator', () => {
+    let events = {};
+
+    return {
+        on(eventName, callback) {
+            events[eventName] = callback;
+        },
+        emit(eventName, params) {
+            if (events[eventName]) {
+                events[eventName](params);
+            }
+        },
+        removeAllListeners() {
+            events = {};
+        },
+    };
+});
```

Comment:
If this simplified synchronous mediator looks useful, we could probably move it to `lib/__mocks__`. I found that providing a synchronous API for the mock means we no longer have to emit events as test hooks.